engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala

```diff
@@ -184,6 +189,23 @@ object ProgramExecutionSupport {
             )
           )
         }
+        executionFrame.cache.getTypes.forEach {
+          (expressionId, expressionType) =>
+            if (!executionFrame.syncState.isExpressionSync(expressionId)) {
```

Comment:
This must be the core of the change, right @4e6? This means that we will now get more `onCachedValueCallback` then before? If not(!?) `syncState.isExpressionSync` we trigger the callback. What does it mean `syncState.isExpressionSync`?